dev/core/src/com/google/gwt/dev/shell/jetty/JettyLauncher.java

```diff
@@ -566,6 +600,15 @@ private WebAppContextWithReload(TreeLogger logger, String webApp,
     protected void doStart() throws Exception {
       setClassLoader(new WebAppClassLoaderExtension());
       super.doStart();
+
+      // After start, warn if a servlet/filter was configured other than those provided by Jetty
+      boolean hasNonJettyFiltersOrServlets = Stream.concat(
+              getServletContext().getServletRegistrations().values().stream(),
+              getServletContext().getFilterRegistrations().values().stream()
+      ).anyMatch(r -> !r.getClassName().startsWith("org.eclipse.jetty"));
```

Comment:
Fwiw, this would exclude explicit registrations of servlets and filters from the Jetty packages (another instance of the default servlet, a QoS or Header filter)

It won't match configurations of security constraints or environmental entries either, as well as ServletContainerInitializers (that could store values in the servlet context, or register e.g. session listeners, without registering servlets and filters).

IIRC, people can also put a `jetty-web.xml` in their `WEB-INF` to configure the Jetty WebAppContext further, that wouldn't be detected here.

So, while it would match the majority of cases, it will miss a few edge-cases.
I wonder if we shouldn't try to match loading of `WEB-INF/web.xml`, `META-INF/web-fragment.xml`, `META-INF/services/javax.servlet.ServletContextInitializer`, `META-INF/jetty-web.xml` (and the `web-jetty.xml` and other variants)
And/or maybe if the `WebAppClassLoaderExtension` loads any class we haven't explicitly allowed? (might not detect the QosFilter or HeaderFilter, or explicit DefaultServlet, so would have to be combined with the other resource-based approach?)

Reply:
While you're correct that we would miss these edge cases on startup, I'll bet that we're still missing cases - that's why any non-404 response (including opening a websocket unless I'm very much mistaken) will also trigger the warning to log (though not right at startup, but in limited testing they appear before the ncsa-style log entries). Does that seem like a sufficient mechanism for catch-all, or do you think we need to get, traverse all of the supported configuration types as well?

Reply:
I was thinking on using the classloader's resource loading for those files I listed. Do you think that would work?
Anyway, better merge this sooner than later so feel free to merge as-is.